crates/router/src/core/payments.rs

```diff
@@ -1224,14 +1225,12 @@ pub fn should_call_connector<Op: Debug, F: Clone>(
                 .is_none()
         }
         "PaymentStatus" => {
-            matches!(
-                payment_data.payment_intent.status,
-                storage_enums::IntentStatus::Processing
-                    | storage_enums::IntentStatus::RequiresCustomerAction
-                    | storage_enums::IntentStatus::RequiresMerchantAction
-                    | storage_enums::IntentStatus::RequiresCapture
-                    | storage_enums::IntentStatus::PartiallyCaptured
-            ) && payment_data.force_sync.unwrap_or(false)
+            payment_data.force_sync.unwrap_or(false)
+                && helpers::check_force_psync_precondition(
+                    &payment_data.payment_attempt,
+                    merchant_account,
+                )
+                .unwrap_or(false)
```

Comment:
check for encoded_data in the payment_data, if encoded data is present then we should allow psync call to connector